docs: add internationalization documentation for DatePicker (#1552)
vaadin/docs: articles/components/date-picker/index.asciidoc

----
--

== Internationalization (i18n)

> Date Picker provides an API for internationalization.

Author: I updated the description and referenced the date format section.

[.example]
--

> [source,html]

Reviewer: ```suggestion
[source,typescript]
```
Let's format this as Typescript code.

----
--

== Internationalization (i18n)

Date Picker allows localizing texts and labels, such as month names and button labels.

> See <<../date-picker#date-format,Date Format>> for combined use.

Reviewer: > combined use.

For me it's not clear what this means (combining what?), would suggest:

```suggestion
See also how to <<#date-format,configure a custom date format>>.
```

It would also make sense to move this sentence below the example to keep the example in the context of the previous sentence.
